app/controllers/api/traces_controller.rb

```diff
@@ -12,7 +12,14 @@ class TracesController < ApiController
     def show
       @trace = Trace.visible.find(params[:id])
 
-      head :forbidden unless @trace.public? || @trace.user == current_user
+      if @trace.public? || @trace.user == current_user
+        respond_to do |format|
+          format.xml { render :content_type => "application/xml" }
+          format.json { render :content_type => "application/json" }
+        end
```

Comment:
I don't think you need the explicit content type here. I think this will suffice:
```suggestion
        respond_to do |format|
          format.xml
          format.json
        end
```